tests/filter/test_unsubscribe_all.py

```diff
@@ -45,14 +45,12 @@ def test_filter_unsubscribe_all_on_peer_with_no_subscription(self):
             self.delete_all_filter_subscriptions({"requestId": "1"})
             raise AssertionError("Unsubscribe all on peer without subscriptions worked!!!")
         except Exception as ex:
-            assert "Not Found" and "peer has no subscriptions" in str(ex)
-
-    def test_filter_unsubscribe_all_with_no_request_id(self, subscribe_main_nodes):
-        try:
-            self.delete_all_filter_subscriptions({})
-            raise AssertionError("Unsubscribe all with no request id worked!!!")
-        except Exception as ex:
-            assert "Bad Request" in str(ex)
+            if self.node2.is_nwaku():
+                assert "Not Found" and "peer has no subscriptions" in str(ex)
```

Comment:
I'd wager this doesn't work as expected.
I believe you need to make the two checks separately, otherwise the first term will be always evaluated to `True` separately.

Reply:
good catch! fixed it